src/DefaultBuilder/src/ConfigureWebHostBuilder.cs

```diff
@@ -54,6 +54,10 @@ public IWebHostBuilder ConfigureAppConfiguration(Action<WebHostBuilderContext, I
                 // We allow changing the web root since it's based off the content root and typically
                 // read after the host is built.
                 _environment.WebRootPath = Path.Combine(_environment.ContentRootPath, value);
+                if (!Directory.Exists(_environment.WebRootPath))
```

Comment:
This is slightly messy but I'm assuming the next line throws without it? You could end up creating multiple directories since this happens before the final build.

Reply:
I think you only want to create it at the end.

Reply:
Yeah, the PhysicalFileProvider will throw if the directory doesn't exist:

https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs#L64-L67

Reply:
I think the question is whether CreateDirectory() throws when the directory does exist. It [doesn't look like it](https://github.com/dotnet/runtime/blob/580d17fd334e297d981fce02753ed7a1ebe8b8eb/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs#L291-L295).

Reply:
Yeah, @davidfowl concern was around the behavior when the user calls `UseSetting` multiple times with different directories. Ultimately, we only respect the last directory they used but with the previous logic we'll end up creating a directory if it doesn't exist for all of them which is cHaOtIC.